packages/@uppy/tus/src/index.js

```diff
@@ -252,6 +259,24 @@ module.exports = class Tus extends BasePlugin {
         resolve(upload)
       }
 
+      uploadOptions.onShouldRetry = (err, retryAttempt, options) => {
+        if (err?.originalResponse?.getStatus() === 429) {
+          if (!this.requests.isPaused) {
+            const { done, value } = this.#retryDelayIterator.next()
+            if (done) {
+              return false
+            }
+            this.requests.pause(value)
+          }
+          queuedRequest = this.requests.run(queueRequest)
+        } else {
+          setTimeout(() => {
+            queuedRequest = this.requests.run(queuedRequest)
```

Comment:
This line is difficult to understand for me. I am reading it as "we run the queuedRequest and the return value is used to overwrite that object". If we need to keep it like that, let's add a comment here, please. I later understood that `queuedRequest` is actually a function, which makes the naming confusing and makes it even more confusing that we overwrite that function then. :)

I would expect something more along the lines of this:

```
setTimeout(() => {
       queuedRequest.run() // then adjusts internal state if required
}, options.retryDelays[retryAttempt])
```

Reply:
That's a very good illustration of the problem you raised in https://github.com/transloadit/uppy/pull/3394#discussion_r777661936, I made a confusion between `queueRequest` and `queuedRequest` 🤦‍♂️